storage/remote/client.go

```diff
@@ -187,3 +190,70 @@ func (c *Client) Read(ctx context.Context, query *prompb.Query) (*prompb.QueryRe
 
 	return resp.Results[0], nil
 }
+
+type labelsResponse struct {
+	Data     []string `json:"data,omitempty"`
+	Error    string   `json:"error,omitempty"`
+	Warnings []string `json:"warnings,omitempty"`
+}
+
+// makeLabelsRequest makes the actual request for /label and /labels/{name}/values from the remote client
+func (c *Client) makeLabelsRequest(ctx context.Context, url string) ([]string, storage.Warnings, error) {
+	httpReq, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "unable to create request")
+	}
+	httpReq.Header.Set("User-Agent", userAgent)
+
+	ctx, cancel := context.WithTimeout(ctx, c.timeout)
+	defer cancel()
+
+	httpResp, err := c.client.Do(httpReq.WithContext(ctx))
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "error sending request")
+	}
+	defer func() {
+		io.Copy(ioutil.Discard, httpResp.Body)
```

Comment:
Pro tip, we can be explicit when we discard errors.

```suggestion
		_, _ io.Copy(ioutil.Discard, httpResp.Body)
```